client/core/bond.go

```diff
@@ -541,19 +575,28 @@ func (c *Core) rotateBonds(ctx context.Context) {
 		}
 
 		if acctBondState.tierChange != 0 {
-			reserveDelta := bondOverlap * -acctBondState.tierChange * int64(bondAsset.Amt)
-			netReserveDeltas[acctBondState.bondAssetID] += reserveDelta // negative means free reserves
-			// After applying the net reserveDelta, we would normally *zero* the
+			// Reserve enough funds to recreate current bonds.
+			reserve := bondOverlap * uint64(acctBondState.liveStrength) * bondAsset.Amt
```

Comment:
The reason for the change is that now the `acctBondState.tierChange` may not be reliable to decide how much to reserve. If bonus tiers go up then we will start reserving 0 even though we need to reserve enough for at least one bond.

I think the original logic was created this way to allow for less reserves as we needed less because of bonus bond tiers, and I attempted to save that logic, but maybe we should just simply reserve target tiers worth?

> The use of a bare liveStrength here is curious.

It was my thinking that we want enough to re-post all live bonds. That is strong and weak. I think strong and pending could also work, but pending, weak, and strong would be overkill, as we are making pending bonds in response to weak bonds so it would count those twice... Would the second return from `bondTotalInternal` be the same as `liveStrength` here?

Reply:
Hmm, should this bit subtracted then from live bonds? bondOverlap indicates 2x current? If it was three then 3x current. Also I guess this does not include fees we should reserve? Can clean this up better after v2 ConnectResult? I'm unclear on exactly how much we want to reserve. Do we want to count bonus tiers? We must count any negative tiers... If we are -1 because of bad behavior, we will always need to reserve target + 1.

Reply:
> I think the original logic was created this way to allow for less reserves as we needed less because of bonus bond tiers, and I attempted to save that logic, but maybe we should just simply reserve target tiers worth?

Maybe, but trading limits scale with **bonded** tier. Bonus tiers are an illusion. I'm not certain why they exist at all, tbh. The server doesn't need to add them. They do nothing for the client as "tiers". Client should be more interested in score as a form of reputation and as a penalty buffer. I may propose removing these bonus tiers if I can get the upgrade logic worked out.

Reply:
Ok. Since I am unsure what the ultimate plan for the bonus tiers is will leave this for now. If you want to absorb usable changes here into #2471 or have a different plan please do. I can remove the reserves changes from this as well.

I don't have a strong opinion on whether the user should have expanded trading limits for good behavior.

Reply:
I guess I was thinking that it should look something like this. https://github.com/buck54321/dcrdex/compare/04116f8...buck54321:dcrdex:reserves-adjustment